libraries/net/node.cpp

```diff
@@ -5045,4 +5165,71 @@ namespace graphene { namespace net { namespace detail {
 
   } // end namespace detail
 
+  /***
+   * @brief Helper to convert a string to a collection of endpoints
+   *
+   * This converts a string (i.e. "bitshares.eu:665535" to a collection of endpoints.
+   * NOTE: Throws an exception if not in correct format or was unable to resolve URL.
+   *
+   * @param in the incoming string
+   * @returns a vector of endpoints
+   */
+  std::vector<fc::ip::endpoint> node::resolve_string_to_ip_endpoints(const std::string& in)
+  {
+     try
+     {
+        std::string::size_type colon_pos = in.find(':');
+        if (colon_pos == std::string::npos)
+           FC_THROW("Missing required port number in endpoint string \"${endpoint_string}\"",
+                    ("endpoint_string", in));
+        std::string port_string = in.substr(colon_pos + 1);
+        try
+        {
+           uint16_t port = boost::lexical_cast<uint16_t>(port_string);
+
+           std::string hostname = in.substr(0, colon_pos);
+           std::vector<fc::ip::endpoint> endpoints = fc::resolve(hostname, port);
+           if (endpoints.empty())
+              FC_THROW_EXCEPTION( fc::unknown_host_exception,
+                                  "The host name can not be resolved: ${hostname}",
+                                  ("hostname", hostname) );
+           return endpoints;
+        }
+        catch (const boost::bad_lexical_cast&)
+        {
+           FC_THROW("Bad port: ${port}", ("port", port_string));
+        }
+     }
+     FC_CAPTURE_AND_RETHROW((in))
+  }
+
+  void node::add_seed_node(const std::string& endpoint_string, bool connect_immediately)
+  {
+     my->add_seed_node(endpoint_string, connect_immediately);
+  }
+
+  /*****
+   * @brief add a list of nodes to seed the p2p network
+   * @param seeds a vector of url strings
+   * @param connect_immediately attempt a connection immediately
+   */
+  void node::add_seed_nodes(std::vector<std::string> seeds, bool connect_immediately)
+  {
+     for(const std::string& endpoint_string : seeds )
+     {
+        try {
+           my->add_seed_node(endpoint_string, connect_immediately);
+        } catch( const fc::exception& e ) {
+           wlog( "caught exception ${e} while adding seed node ${endpoint}",
+                    ("e", e.to_detail_string())("endpoint", endpoint_string) );
+        }
+     }
+
+  }
+
+  void node::set_advertise_algorithm( std::string algo, std::vector<std::string> advertise_list )
```

Comment:
How about changing the 2nd parameter type to `optional` or a pointer?

Reply:
I went ahead and made it an `fc::optional`. I assume that is to save an allocation of a vector when the parameter is not passed and the default is used. If this is for another reason, would you please enlighten me?